src/checks/#cc4a#modern_language.clas.abap

```diff
@@ -0,0 +1,1157 @@
+CLASS /cc4a/modern_language DEFINITION
+  PUBLIC
+  FINAL
+  CREATE PUBLIC .
+
+  PUBLIC SECTION.
+
+    INTERFACES if_ci_atc_check .
+    CONSTANTS c_code_move TYPE if_ci_atc_check=>ty_finding_code VALUE 'MOVE'.
+    CONSTANTS c_code_translate TYPE if_ci_atc_check=>ty_finding_code VALUE 'TRANSLATE'.
+    CONSTANTS c_code_line_exists TYPE if_ci_atc_check=>ty_finding_code VALUE 'LINE_EXIST'.
+    CONSTANTS c_code_prefer_new TYPE if_ci_atc_check=>ty_finding_code VALUE 'PREFER_NEW'.
+    CONSTANTS c_code_call_method TYPE if_ci_atc_check=>ty_finding_code VALUE 'CALL_METH'.
+    CONSTANTS c_code_method_exporting TYPE if_ci_atc_check=>ty_finding_code VALUE 'METH_EXP'.
+    CONSTANTS c_code_exporting_receiving TYPE if_ci_atc_check=>ty_finding_code VALUE 'EXP_REC'.
+    CONSTANTS c_code_text_assembly TYPE if_ci_atc_check=>ty_finding_code VALUE 'TEXT_ASM'.
+    CONSTANTS c_code_qf_synerr TYPE if_ci_atc_check=>ty_finding_code VALUE 'QF_SYNERR'.
+    CONSTANTS c_code_qf_tsterr TYPE if_ci_atc_check=>ty_finding_code VALUE 'QF_TSTERR'.
+    CONSTANTS c_qf_move TYPE cl_ci_atc_quickfixes=>ty_quickfix_code VALUE 'QF_MOVE'.
+    CONSTANTS c_qf_translate TYPE cl_ci_atc_quickfixes=>ty_quickfix_code VALUE 'QF_TRANSL'.
+    CONSTANTS c_qf_line_exists TYPE cl_ci_atc_quickfixes=>ty_quickfix_code VALUE 'QF_LINEEX'.
+    CONSTANTS c_qf_prefer_new TYPE cl_ci_atc_quickfixes=>ty_quickfix_code VALUE 'QF_PREFNEW'.
+    CONSTANTS c_qf_call_method TYPE cl_ci_atc_quickfixes=>ty_quickfix_code VALUE 'QF_CALLM'.
+    CONSTANTS c_qf_method_exporting TYPE cl_ci_atc_quickfixes=>ty_quickfix_code VALUE 'QF_MEXP'.
+    CONSTANTS c_qf_exporting_receiving TYPE cl_ci_atc_quickfixes=>ty_quickfix_code VALUE 'QF_EXP_REC'.
+    CONSTANTS c_qf_text_assembly TYPE cl_ci_atc_quickfixes=>ty_quickfix_code VALUE 'QF_TEXTASM'.
+    CONSTANTS c_ps_deprecated_key TYPE string VALUE 'DEPRECATED_KEY'.
+    CONSTANTS c_ps_line_exists TYPE string VALUE 'PREF_LINE_EX'.
+    CONSTANTS c_ps_prefer_new TYPE string VALUE 'PREF_NEW'.
+    CONSTANTS c_ps_call_method TYPE string VALUE 'CALL_METH_USAGE'.
+    CONSTANTS c_ps_method_exporting TYPE string VALUE 'OPTL_EXP'.
+    CONSTANTS c_ps_exporting_receiving TYPE string VALUE 'RECEIVING_USAGE'.
+    CONSTANTS c_ps_text_assembly TYPE string VALUE 'TEXT_ASSEMBLY'.
+    CONSTANTS c_max_line_length TYPE i VALUE 255.
```

Comment:
1. Please don't use hungarian notation (cf. [style guide](https://github.com/SAP/styleguides/blob/main/clean-abap/CleanABAP.md#avoid-encodings-esp-hungarian-notation-and-prefixes))
2. Try to be consistent with the other checks structure their constants for finding codes and use structured constants for the message codes and quickfix codes like e.g. [here](https://github.com/SAP/code-pal-for-abap-cloud/blob/a28cadc0ef08896aa69bc5e9f04d84eb8527e5a1/src/checks/%23cc4a%23method_signature.clas.abap#L10)
3. Do all these constants need to be public? E.g. `c_max_line_length` doesn't seem useful to external callers (and if we need a constant for the ABAP line length its correct place is probably in `/CC4A/IF_ABAP_ANALYZER`).